Neos.ContentRepository.Core/Classes/Service/ContentRepositoryMaintainer.php

```diff
@@ -0,0 +1,200 @@
+        }
+        $bootResult = $this->subscriptionEngine->boot(progressCallback: $progressCallback);
+        if ($bootResult->errors !== null) {
+            return self::createErrorForReason('catchup', $bootResult->errors);
+        }
+        return null;
+    }
+
+    /**
+     * Catchup one specific projection.
+     *
+     * The explicit catchup is required for new projections in the booting state.
+     *
+     * We don't offer an API to catch up all projections catchupAllProjection as we would have to distinct between booting or catchup if its active already.
+     *
+     * This method is only needed in rare cases for debugging or after installing a new projection or fixing its errors.
+     */
+    public function catchupProjection(SubscriptionId $subscriptionId, \Closure|null $progressCallback = null): Error|null
+    {
+        $bootResult = $this->subscriptionEngine->boot(SubscriptionEngineCriteria::create([$subscriptionId]), progressCallback: $progressCallback);
+        if ($bootResult->errors !== null) {
+            return self::createErrorForReason('catchup', $bootResult->errors);
+        }
+        if ($bootResult->numberOfProcessedEvents > 0) {
+            // the projection was bootet
+            return null;
+        }
+        // todo the projection was active, and we might still want to catch it up ... find reason for this? And combine boot and catchup?
+        $catchupResult = $this->subscriptionEngine->catchUpActive(SubscriptionEngineCriteria::create([$subscriptionId]), progressCallback: $progressCallback);
+        if ($catchupResult->errors !== null) {
+            return self::createErrorForReason('catchup', $catchupResult->errors);
+        }
+        return null;
+    }
+
+    /**
+     * WARNING: Removes all events from the content repository and resets the projections
+     * This operation cannot be undone.
+     */
+    public function prune(): Error|null
+    {
+        // prune all streams:
+        foreach ($this->findAllContentStreamStreamNames() as $contentStreamStreamName) {
+            $this->eventStore->deleteStream($contentStreamStreamName);
+        }
+        foreach ($this->findAllWorkspaceStreamNames() as $workspaceStreamName) {
+            $this->eventStore->deleteStream($workspaceStreamName);
+        }
+        $resetResult = $this->subscriptionEngine->reset();
+        if ($resetResult->errors !== null) {
+            return self::createErrorForReason('reset', $resetResult->errors);
+        }
+        // todo reintroduce skipBooting flag to reset
+        $bootResult = $this->subscriptionEngine->boot();
+        if ($bootResult->errors !== null) {
+            return self::createErrorForReason('booting', $bootResult->errors);
+        }
+        return null;
+    }
+
+    private static function createErrorForReason(string $method, Errors $errors): Error
+    {
+        // todo log throwable via flow???, but we are here in the CORE ...
```

Comment:
is that important after all for cli? Or can we just swallow it and print the nice human readable message.
It will be persistent in the db after all if it was a projection error.